Project-Hotpot/Project-Hotpot/Managers/SpotifyManager.swift

```diff
@@ -146,6 +145,40 @@ extension SpotifyManager: SPTSessionManagerDelegate {
 
 // MARK: - Networking
 extension SpotifyManager {
+    
+    func makeSpotifyRequest (request: URLRequest, completion: @escaping ([String: Any]?, Error?) -> Void) {
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+            guard let data = data,                              // is there data
+                  let response = response as? HTTPURLResponse,  // is there HTTP response
+                  (200 ..< 300) ~= response.statusCode,         // is statusCode 2XX
+                  error == nil else {                           // was there no error, otherwise ...
+                NSLog("Error fetching token \(error?.localizedDescription ?? "")")
+                return completion(nil, error)
+            }
+            let responseObject = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
+            completion(responseObject, nil)
+        }
+        task.resume()
+        
+    }
+    
+    func fetchSongsFromGenre(genre:String, offset: Int, completion: @escaping ([String: Any]?, Error?) -> Void) {
+        
+        guard let url = URL(string: "https://api.spotify.com/v1/search?q=" + "genre:" + genre + "&type=track&limit=50&offset=" + String(offset)),
+              let accessToken = self.appRemote.connectionParameters.accessToken
+        else {
+            NSLog("failed fetching song from genre, nil access token")
+            return
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        let authorizationValue = "Bearer " + accessToken
+        request.allHTTPHeaderFields = ["Authorization": authorizationValue,
+                                       "Content-Type": "application/json",
+                                       "Accept": "application/json"]
```

Comment:
My logic for leaving this within the methods is since the fetchAccessToken method uses the makeSpotifyRequest method as well, but requires different HTTPHeaderFields. Since this varies from method to method, I thought this was the most efficient way to consolidate code.

Reply:
Got it for such cases it is always advised to use a RequestBuilder kind of object  to minimize rewriting a lot of network code so you could do something like:
`
let request = RequestBuilder().method(GET).url(url).token(blahblah).header(key, value).build()
makeSpotifyCall(request,...)
`
This is a widely known design pattern "Builder Pattern" and you can find examples all over the internet. e.g. - https://betterprogramming.pub/implement-the-builder-design-pattern-in-swift-5-ff5bc6f2fc3d